src/styles/main.scss

```diff
@@ -1,3 +1,69 @@
+@import "./utils/variables";
+@import "./utils/function";
+
 body {
   margin: 0;
 }
+
+.container {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: get-width($ceil-width, $gap);
+  padding: 0;
+
+  &__item {
+    position: relative;
+    box-sizing: border-box;
+    height: $ceil-height;
+    width: $ceil-width;
+    background-color: #eee;
+    list-style: none;
+    transition: 0.5s ease;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  }
+}
+
+@for $index from 1 to 32 {
+  .calendar__item:nth-child(#{$index})::before {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    font-family: Arial, sans-serif;
+    font-size: 30px;
+    text-align: center;
+    line-height: $ceil-height;
+    content: "#{$index}";
+  }
+}
+
+@for $i from 28 to 31 {
+  .calendar--month-length_#{$i} {
+
+    @for $j from $i to 31 {
+      .calendar__item:nth-child(#{$j+1}) {
+        display: none;
+      }
+    }
+  }
+}
```

Comment:
let's use one loop to optimise the code and use meaningful variable names instead of `i` and `j`